crates/task_executor/src/executor.rs

```diff
@@ -0,0 +1,27 @@
+use std::future::Future;
+
+/// An abstraction for executing tasks, suitable for both CPU-bound and I/O-bound operations.
+pub trait TaskExecutor {
+    type SpawnBlockingError;
+    type SpawnError;
```

Comment:
When we have client code that will be specific to the concrete types of these errors, it may not be trivial to change with another implementation.